### ISSUE TITLE
Add options to setup_waterlevel_forcing for user friendliness

### ISSUE THREAD

**Issue by Leynse** [Deltares/hydromt_sfincs#222]
### Kind of request

Adding new functionality

### Enhancement Description

Add options like:
- Buffer = None option so automatically all provided points are used
- Consider changing the default to this setting (by default use all points, not limit to 5000 m), otherwise increase default buffer value
- Add more clear error message for the case no points are found > error messages now come from hydromt core and are very unclear
- e.g. message: 'no points found within buffer'

### Use case

See above

**Comment by roeldegoede** [comment]
Agree, this originates from starting with a GTSM like dataset where you want to filter out a lot of points, however it now also filters points when we only provide a few (that can be annoyting, but might be worth a warning (?) if they're far away from the domain). If we change the default, we somehow would like to give a warning I guess when people use a global dataset .. 

Let's discuss the preferred option

**Comment by roeldegoede** [comment]
I didn't see a neat way of implementing the buffer=None option. Think it would end up in some more if-else statements which would not improve code readability.

A clear warning is now provided, hope that is sufficient. Increasing the buffer to very large values should results in all provided points to be added.
